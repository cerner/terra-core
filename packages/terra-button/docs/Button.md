| Prop Name | Type | Is Required | Default Value | Description | 
|-|-|-|-|-|
| children | `node`  | optional |  | Child Nodes |
| href | `string`  | optional |  | Sets the href. When set will render the component as an anchor tag |
| icon | `element`  | optional |  | An optional icon. Nested inline with the text when provided |
| isBlock | `bool`  | optional | `false` | Whether or not the button should display as a block |
| isCompact | `bool`  | optional | `false` | Whether or not the button has reduced padding |
| isDisabled | `bool`  | optional | `false` | Whether or not the button should be disabled |
| isReversed | `bool`  | optional | `false` | Reverses the position of the icon and text |
| onClick | `func`  | optional |  | Callback function triggered when clicked |
| size | `enum('tiny','small','medium','large','huge')`  | optional |  | Sets the button size. One of tiny, small, medium, large, huge |
| text | `string`  | optional |  | Sets the button text |
| variant | `enum('default','link','primary','secondary')`  | optional | `'default'` | Sets the button variant. One of primary, secondary, or link |