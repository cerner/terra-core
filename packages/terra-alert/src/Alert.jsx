import React from 'react';
import PropTypes from 'prop-types';
import ResponsiveElement from 'terra-responsive-element';
import Button from 'terra-button';
import IconAlert from 'terra-icon/lib/icon/IconAlert';
import IconError from 'terra-icon/lib/icon/IconError';
import IconWarning from 'terra-icon/lib/icon/IconWarning';
import IconInformation from 'terra-icon/lib/icon/IconInformation';
import IconSuccess from 'terra-icon/lib/icon/IconSuccess';
import classNames from 'classnames/bind';
import styles from './Alert.module.scss';

const cx = classNames.bind(styles);

const AlertTypes = {
  ALERT: 'alert',
  ERROR: 'error',
  WARNING: 'warning',
  ADVISORY: 'advisory',
  INFO: 'info',
  SUCCESS: 'success',
  CUSTOM: 'custom',
};

const propTypes = {
  /**
   * An action element to be added to the action section of the alert.
   */
  action: PropTypes.element,
  /**
   * Child Nodes providing the message content for the alert. Can contain text and HTML.
   */
  children: PropTypes.node,
  /**
   * The icon to be used for an alert of type custom. This will not be used for any other alert types.
   */
  customIcon: PropTypes.element,
  /**
   * The status bar color to be used for an alert of type custom.
   */
  customStatusColor: PropTypes.string,
  /**
   * Callback function triggered when Dismiss button is clicked. The presence of this prop will cause the Dismiss button to be included on the alert.
   */
  onDismiss: PropTypes.func,
  /**
   * The title for the alert which will be bolded.
   */
  title: PropTypes.string,
  /**
   * The type of alert to be rendered. One of `alert`, `error`, `warning`, `advisory`,
   * `info`, `success`, `custom`.
   */
  type: PropTypes.oneOf([
    AlertTypes.ALERT,
    AlertTypes.ERROR,
    AlertTypes.WARNING,
    AlertTypes.ADVISORY,
    AlertTypes.INFO,
    AlertTypes.SUCCESS,
    AlertTypes.CUSTOM,
  ]),
};

const defaultProps = {
  action: null,
  children: '',
  customIcon: null,
  customStatusColor: '',
  onDismiss: null,
  title: '',
  type: AlertTypes.ALERT,
};

const getAlertIcon = (type, customIcon) => {
  switch (type) {
    case AlertTypes.ALERT:
      return (<span className={cx('icon')}><IconAlert /></span>);
    case AlertTypes.ERROR:
      return (<span className={cx('icon')}><IconError /></span>);
    case AlertTypes.WARNING:
      return (<span className={cx('icon')}><IconWarning /></span>);
    case AlertTypes.ADVISORY:
      return null;
    case AlertTypes.INFO:
      return (<span className={cx('icon')}><IconInformation /></span>);
    case AlertTypes.SUCCESS:
      return (<span className={cx('icon')}><IconSuccess /></span>);
    case AlertTypes.CUSTOM:
      return (<span className={cx('icon')}>{customIcon}</span>);
    default:
      return null;
  }
};

const contextTypes = {
  /* eslint-disable consistent-return */
  intl: (context) => {
    if (context.intl === undefined) {
      return new Error('Component is internationalized, and must be wrapped in terra-base');
    }
  },
};

const Alert = ({
  type,
  children,
  title,
  customIcon,
  customStatusColor,
  onDismiss,
  action,
  ...customProps
}, {
  intl,
}) => {
  const defaultTitle = type === AlertTypes.CUSTOM ? '' : intl.formatMessage({ id: `Terra.alert.${type}` });
  const attributes = Object.assign({}, customProps);
  const narrowAlertClassNames = cx([
    'alert-base',
    type,
    'narrow',
    attributes.className,
  ]);
  const wideAlertClassNames = cx([
    'alert-base',
    type,
    'wide',
    attributes.className,
  ]);

  let actionsSection = '';
  let dismissButton = '';
  const outerDivStyle = {};
  let alertSectionClassName = cx('section');
  let actionsClassName = cx('actions');
  let bodyClassNameForNarrowParent = cx(['body', 'body-std']);

  if (type === AlertTypes.CUSTOM) {
    // For custom alert, there is no color assigned to the box-shadow style since it is to be specified
    // in the customStatusColor prop.  The box-shadow style is defined in CSS in order to get the
    // bidirectionality via the mixin.  As per the W3C spec if the box-shadow does not have the color
    // defined, it will use the prevailing color style, so setting that here. But then we need to set
    // the color style for the alert content so that it doesn't pick up the custom status color. We
    // will allow the icon to pick up the color style so that Terra icons will match the color of the
    // status bar.
    outerDivStyle.color = customStatusColor;
    alertSectionClassName = cx(['section', 'section-custom']);
    actionsClassName = cx(['actions', 'actions-custom']);
  }

  if (onDismiss) {
    dismissButton = (<Button text={intl.formatMessage({ id: 'Terra.alert.dismiss' })} onClick={onDismiss} />);
  }
  if (onDismiss || action) {
    bodyClassNameForNarrowParent = cx(['body', 'body-narrow']);
    actionsSection = (
      <div className={actionsClassName}>
        {action}
        {dismissButton}
      </div>
    );
  }

  const alertMessageContent = (
    <div className={alertSectionClassName}>
      {(title || defaultTitle) && <strong className={cx('title')}>{title || defaultTitle}</strong>}
      {children}
    </div>
  );

  return (
    <ResponsiveElement
      tiny={(
        <div {...attributes} className={narrowAlertClassNames} style={outerDivStyle}>
          <div className={bodyClassNameForNarrowParent}>
            {getAlertIcon(type, customIcon)}
            {alertMessageContent}
          </div>
          {actionsSection}
        </div>
)}
      small={(
        <div {...attributes} className={wideAlertClassNames} style={outerDivStyle}>
          <div className={cx(['body', 'body-std'])}>
            {getAlertIcon(type, customIcon)}
            {alertMessageContent}
          </div>
          {actionsSection}
        </div>
)}
    />

  );
};

Alert.propTypes = propTypes;
Alert.defaultProps = defaultProps;
Alert.contextTypes = contextTypes;
Alert.Opts = {};
Alert.Opts.Types = AlertTypes;

export default Alert;
