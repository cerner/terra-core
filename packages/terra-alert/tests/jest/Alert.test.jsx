/* eslint-disable no-console */
import React from 'react';
/* eslint-disable-next-line import/no-extraneous-dependencies */
import { mountWithIntl } from 'terra-enzyme-intl';

import IconHelp from 'terra-icon/lib/icon/IconHelp';
import Button from 'terra-button';
import Alert from '../../src/Alert';


describe('Alert with no props', () => {
  // Snapshot Tests
  it('should render a default component', () => {
    const wrapper = mountWithIntl(<Alert />);
    expect(wrapper).toMatchSnapshot();
  });

  it('throws error on missing locale prop in Base', () => {
    global.console = { error: jest.fn() };

    expect(() => {
      render(<Alert />);
    }).toThrowError();
    expect(console.error).toBeCalledWith(expect.stringContaining('Component is internationalized, and must be wrapped in terra-base'));
  });
});

describe('Dismissible Alert that includes actions section', () => {
  // Snapshot Tests
  it('should render an alert component with a dismiss button', () => {
    const wrapper = mountWithIntl(<Alert onDismiss={() => { }}>This is a test</Alert>);
    expect(wrapper).toMatchSnapshot();
  });
});

describe('Alert of type alert with text content', () => {
  // Snapshot Tests
  it('should render an Alert component of type alert', () => {
    const wrapper = mountWithIntl(<Alert type={Alert.Opts.Types.ALERT}>This is a test</Alert>);
    expect(wrapper).toMatchSnapshot();
  });
});

describe('Alert of type error with text content', () => {
  // Snapshot Tests
  it('should render an Alert component of type error', () => {
    const wrapper = mountWithIntl(<Alert type={Alert.Opts.Types.ERROR}>This is an error.</Alert>);
    expect(wrapper).toMatchSnapshot();
  });
});

describe('Alert of type warning with text content', () => {
  // Snapshot Tests
  it('should render an Alert component of type warning', () => {
    const wrapper = mountWithIntl(<Alert type={Alert.Opts.Types.WARNING}>This is an warning.</Alert>);
    expect(wrapper).toMatchSnapshot();
  });
});

describe('Alert of type advisory with text content', () => {
  // Snapshot Tests
  it('should render an Alert component of type advisory', () => {
    const wrapper = mountWithIntl(<Alert type={Alert.Opts.Types.ADVISORY}>This is an advisory alert.</Alert>);
    expect(wrapper).toMatchSnapshot();
  });
});

describe('Alert of type info with text content', () => {
  // Snapshot Tests
  it('should render an Alert component of type info', () => {
    const wrapper = mountWithIntl(<Alert type={Alert.Opts.Types.INFO}>This is an information alert.</Alert>);
    expect(wrapper).toMatchSnapshot();
  });
});

describe('Alert of type success with text content', () => {
  // Snapshot Tests
  it('should render an Alert component of type success', () => {
    const wrapper = mountWithIntl(<Alert type={Alert.Opts.Types.SUCCESS}>This is a success alert.</Alert>);
    expect(wrapper).toMatchSnapshot();
  });
});

describe('Alert of type custom with custom title and text content', () => {
  // Snapshot Tests
  it('should render an Alert component of type custom', () => {
    const wrapper = mountWithIntl(<Alert type={Alert.Opts.Types.CUSTOM} title="Help!" customIcon={<IconHelp />} customStatusColor="orange">This is a custom alert.</Alert>);
    expect(wrapper).toMatchSnapshot();
  });
});

describe('Alert of type info with custom title and HTML content', () => {
  // Snapshot Tests
  it('should render an Alert component of type info with custom title and HTML content', () => {
    const wrapper = mountWithIntl(<Alert type={Alert.Opts.Types.INFO} title="Gettysburg Address"><span>Four score and seven years ago . . .</span></Alert>);
    expect(wrapper).toMatchSnapshot();
  });
});

describe('Alert of type success with an action button text content', () => {
  // Snapshot Tests
  it('should render an Alert component of type success with an action button', () => {
    const wrapper = mountWithIntl(<Alert type={Alert.Opts.Types.SUCCESS} action={<Button text="Action" variant={Button.Opts.Variants.EMPHASIS} onClick={() => { }} />}>This is a success alert.</Alert>);
    expect(wrapper).toMatchSnapshot();
  });
});

describe('Dismissable Alert of type custom with action button, custom title and text content', () => {
  // Snapshot Tests
  it('should render an Alert component of type custom with an action button', () => {
    const wrapper = mountWithIntl(<Alert type={Alert.Opts.Types.CUSTOM} onDismiss={() => { }} title="Help!" customIcon={<IconHelp />} customStatusColor="orange" action={<Button text="Action" variant={Button.Opts.Variants.EMPHASIS} onClick={() => { }} />}>This is a custom alert.</Alert>);
    expect(wrapper).toMatchSnapshot();
  });
});
