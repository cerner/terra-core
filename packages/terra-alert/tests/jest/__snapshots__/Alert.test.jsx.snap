// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Alert of type advisory with text content should render an Alert component of type advisory 1`] = `
<IntlProvider
  locale="en-US"
  messages={
    Object {
      "Terra.alert.advisory": "Advisory.",
      "Terra.alert.alert": "Alert.",
      "Terra.alert.dismiss": "Dismiss",
      "Terra.alert.error": "Error.",
      "Terra.alert.info": "Information.",
      "Terra.alert.success": "Success.",
      "Terra.alert.warning": "Warning.",
    }
  }
>
  <Alert
    action={null}
    customIcon={null}
    customStatusColor=""
    onDismiss={null}
    title=""
    type="advisory"
  >
    <ResponsiveElement
      defaultElement={
        <div
          className="alert-base advisory narrow"
          style={Object {}}
        >
          <div
            className="body body-std"
          >
            <div
              className="section"
            >
              <strong
                className="title"
              >
                Advisory.
              </strong>
              This is an advisory alert.
            </div>
          </div>
          
        </div>
      }
      responsiveTo="parent"
      tiny={
        <div
          className="alert-base advisory wide"
          style={Object {}}
        >
          <div
            className="body body-std"
          >
            <div
              className="section"
            >
              <strong
                className="title"
              >
                Advisory.
              </strong>
              This is an advisory alert.
            </div>
          </div>
          
        </div>
      }
    >
      <div />
    </ResponsiveElement>
  </Alert>
</IntlProvider>
`;

exports[`Alert of type alert with text content should render an Alert component of type alert 1`] = `
<IntlProvider
  locale="en-US"
  messages={
    Object {
      "Terra.alert.advisory": "Advisory.",
      "Terra.alert.alert": "Alert.",
      "Terra.alert.dismiss": "Dismiss",
      "Terra.alert.error": "Error.",
      "Terra.alert.info": "Information.",
      "Terra.alert.success": "Success.",
      "Terra.alert.warning": "Warning.",
    }
  }
>
  <Alert
    action={null}
    customIcon={null}
    customStatusColor=""
    onDismiss={null}
    title=""
    type="alert"
  >
    <ResponsiveElement
      defaultElement={
        <div
          className="alert-base alert narrow"
          style={Object {}}
        >
          <div
            className="body body-std"
          >
            <span
              className="icon"
            >
              <IconAlert
                viewBox="0 0 48 48"
                xmlns="http://www.w3.org/2000/svg"
              />
            </span>
            <div
              className="section"
            >
              <strong
                className="title"
              >
                Alert.
              </strong>
              This is an alert.
            </div>
          </div>
          
        </div>
      }
      responsiveTo="parent"
      tiny={
        <div
          className="alert-base alert wide"
          style={Object {}}
        >
          <div
            className="body body-std"
          >
            <span
              className="icon"
            >
              <IconAlert
                viewBox="0 0 48 48"
                xmlns="http://www.w3.org/2000/svg"
              />
            </span>
            <div
              className="section"
            >
              <strong
                className="title"
              >
                Alert.
              </strong>
              This is an alert.
            </div>
          </div>
          
        </div>
      }
    >
      <div />
    </ResponsiveElement>
  </Alert>
</IntlProvider>
`;

exports[`Alert of type custom with custom title and text content should render an Alert component of type custom 1`] = `
<IntlProvider
  locale="en-US"
  messages={
    Object {
      "Terra.alert.advisory": "Advisory.",
      "Terra.alert.alert": "Alert.",
      "Terra.alert.dismiss": "Dismiss",
      "Terra.alert.error": "Error.",
      "Terra.alert.info": "Information.",
      "Terra.alert.success": "Success.",
      "Terra.alert.warning": "Warning.",
    }
  }
>
  <Alert
    action={null}
    customIcon={
      <IconHelp
        viewBox="0 0 48 48"
        xmlns="http://www.w3.org/2000/svg"
      />
    }
    customStatusColor="orange"
    onDismiss={null}
    title="Help!"
    type="custom"
  >
    <ResponsiveElement
      defaultElement={
        <div
          className="alert-base custom narrow"
          style={
            Object {
              "color": "orange",
            }
          }
        >
          <div
            className="body body-std"
          >
            <span
              className="icon"
            >
              <IconHelp
                viewBox="0 0 48 48"
                xmlns="http://www.w3.org/2000/svg"
              />
            </span>
            <div
              className="section section-custom"
            >
              <strong
                className="title"
              >
                Help!
              </strong>
              This is a custom alert.
            </div>
          </div>
          
        </div>
      }
      responsiveTo="parent"
      tiny={
        <div
          className="alert-base custom wide"
          style={
            Object {
              "color": "orange",
            }
          }
        >
          <div
            className="body body-std"
          >
            <span
              className="icon"
            >
              <IconHelp
                viewBox="0 0 48 48"
                xmlns="http://www.w3.org/2000/svg"
              />
            </span>
            <div
              className="section section-custom"
            >
              <strong
                className="title"
              >
                Help!
              </strong>
              This is a custom alert.
            </div>
          </div>
          
        </div>
      }
    >
      <div />
    </ResponsiveElement>
  </Alert>
</IntlProvider>
`;

exports[`Alert of type error with text content should render an Alert component of type error 1`] = `
<IntlProvider
  locale="en-US"
  messages={
    Object {
      "Terra.alert.advisory": "Advisory.",
      "Terra.alert.alert": "Alert.",
      "Terra.alert.dismiss": "Dismiss",
      "Terra.alert.error": "Error.",
      "Terra.alert.info": "Information.",
      "Terra.alert.success": "Success.",
      "Terra.alert.warning": "Warning.",
    }
  }
>
  <Alert
    action={null}
    customIcon={null}
    customStatusColor=""
    onDismiss={null}
    title=""
    type="error"
  >
    <ResponsiveElement
      defaultElement={
        <div
          className="alert-base error narrow"
          style={Object {}}
        >
          <div
            className="body body-std"
          >
            <span
              className="icon"
            >
              <IconError
                viewBox="0 0 48 48"
                xmlns="http://www.w3.org/2000/svg"
              />
            </span>
            <div
              className="section"
            >
              <strong
                className="title"
              >
                Error.
              </strong>
              This is an error.
            </div>
          </div>
          
        </div>
      }
      responsiveTo="parent"
      tiny={
        <div
          className="alert-base error wide"
          style={Object {}}
        >
          <div
            className="body body-std"
          >
            <span
              className="icon"
            >
              <IconError
                viewBox="0 0 48 48"
                xmlns="http://www.w3.org/2000/svg"
              />
            </span>
            <div
              className="section"
            >
              <strong
                className="title"
              >
                Error.
              </strong>
              This is an error.
            </div>
          </div>
          
        </div>
      }
    >
      <div />
    </ResponsiveElement>
  </Alert>
</IntlProvider>
`;

exports[`Alert of type info with custom title and HTML content should render an Alert component of type info with custom title and HTML content 1`] = `
<IntlProvider
  locale="en-US"
  messages={
    Object {
      "Terra.alert.advisory": "Advisory.",
      "Terra.alert.alert": "Alert.",
      "Terra.alert.dismiss": "Dismiss",
      "Terra.alert.error": "Error.",
      "Terra.alert.info": "Information.",
      "Terra.alert.success": "Success.",
      "Terra.alert.warning": "Warning.",
    }
  }
>
  <Alert
    action={null}
    customIcon={null}
    customStatusColor=""
    onDismiss={null}
    title="Gettysburg Address"
    type="info"
  >
    <ResponsiveElement
      defaultElement={
        <div
          className="alert-base info narrow"
          style={Object {}}
        >
          <div
            className="body body-std"
          >
            <span
              className="icon"
            >
              <IconInformation
                viewBox="0 0 48 48"
                xmlns="http://www.w3.org/2000/svg"
              />
            </span>
            <div
              className="section"
            >
              <strong
                className="title"
              >
                Gettysburg Address
              </strong>
              <span>
                Four score and seven years ago . . .
              </span>
            </div>
          </div>
          
        </div>
      }
      responsiveTo="parent"
      tiny={
        <div
          className="alert-base info wide"
          style={Object {}}
        >
          <div
            className="body body-std"
          >
            <span
              className="icon"
            >
              <IconInformation
                viewBox="0 0 48 48"
                xmlns="http://www.w3.org/2000/svg"
              />
            </span>
            <div
              className="section"
            >
              <strong
                className="title"
              >
                Gettysburg Address
              </strong>
              <span>
                Four score and seven years ago . . .
              </span>
            </div>
          </div>
          
        </div>
      }
    >
      <div />
    </ResponsiveElement>
  </Alert>
</IntlProvider>
`;

exports[`Alert of type info with text content should render an Alert component of type info 1`] = `
<IntlProvider
  locale="en-US"
  messages={
    Object {
      "Terra.alert.advisory": "Advisory.",
      "Terra.alert.alert": "Alert.",
      "Terra.alert.dismiss": "Dismiss",
      "Terra.alert.error": "Error.",
      "Terra.alert.info": "Information.",
      "Terra.alert.success": "Success.",
      "Terra.alert.warning": "Warning.",
    }
  }
>
  <Alert
    action={null}
    customIcon={null}
    customStatusColor=""
    onDismiss={null}
    title=""
    type="info"
  >
    <ResponsiveElement
      defaultElement={
        <div
          className="alert-base info narrow"
          style={Object {}}
        >
          <div
            className="body body-std"
          >
            <span
              className="icon"
            >
              <IconInformation
                viewBox="0 0 48 48"
                xmlns="http://www.w3.org/2000/svg"
              />
            </span>
            <div
              className="section"
            >
              <strong
                className="title"
              >
                Information.
              </strong>
              This is an information alert.
            </div>
          </div>
          
        </div>
      }
      responsiveTo="parent"
      tiny={
        <div
          className="alert-base info wide"
          style={Object {}}
        >
          <div
            className="body body-std"
          >
            <span
              className="icon"
            >
              <IconInformation
                viewBox="0 0 48 48"
                xmlns="http://www.w3.org/2000/svg"
              />
            </span>
            <div
              className="section"
            >
              <strong
                className="title"
              >
                Information.
              </strong>
              This is an information alert.
            </div>
          </div>
          
        </div>
      }
    >
      <div />
    </ResponsiveElement>
  </Alert>
</IntlProvider>
`;

exports[`Alert of type success with an action button text content should render an Alert component of type success with an action button 1`] = `
<IntlProvider
  locale="en-US"
  messages={
    Object {
      "Terra.alert.advisory": "Advisory.",
      "Terra.alert.alert": "Alert.",
      "Terra.alert.dismiss": "Dismiss",
      "Terra.alert.error": "Error.",
      "Terra.alert.info": "Information.",
      "Terra.alert.success": "Success.",
      "Terra.alert.warning": "Warning.",
    }
  }
>
  <Alert
    action={
      <Button
        isBlock={false}
        isCompact={false}
        isDisabled={false}
        isIconOnly={false}
        isReversed={false}
        onClick={[Function]}
        text="Action"
        type="button"
        variant="emphasis"
      />
    }
    customIcon={null}
    customStatusColor=""
    onDismiss={null}
    title=""
    type="success"
  >
    <ResponsiveElement
      defaultElement={
        <div
          className="alert-base success narrow"
          style={Object {}}
        >
          <div
            className="body body-narrow"
          >
            <span
              className="icon"
            >
              <IconSuccess
                viewBox="0 0 48 48"
                xmlns="http://www.w3.org/2000/svg"
              />
            </span>
            <div
              className="section"
            >
              <strong
                className="title"
              >
                Success.
              </strong>
              This is a success alert.
            </div>
          </div>
          <div
            className="actions"
          >
            <Button
              isBlock={false}
              isCompact={false}
              isDisabled={false}
              isIconOnly={false}
              isReversed={false}
              onClick={[Function]}
              text="Action"
              type="button"
              variant="emphasis"
            />
            
          </div>
        </div>
      }
      responsiveTo="parent"
      tiny={
        <div
          className="alert-base success wide"
          style={Object {}}
        >
          <div
            className="body body-std"
          >
            <span
              className="icon"
            >
              <IconSuccess
                viewBox="0 0 48 48"
                xmlns="http://www.w3.org/2000/svg"
              />
            </span>
            <div
              className="section"
            >
              <strong
                className="title"
              >
                Success.
              </strong>
              This is a success alert.
            </div>
          </div>
          <div
            className="actions"
          >
            <Button
              isBlock={false}
              isCompact={false}
              isDisabled={false}
              isIconOnly={false}
              isReversed={false}
              onClick={[Function]}
              text="Action"
              type="button"
              variant="emphasis"
            />
            
          </div>
        </div>
      }
    >
      <div />
    </ResponsiveElement>
  </Alert>
</IntlProvider>
`;

exports[`Alert of type success with text content should render an Alert component of type success 1`] = `
<IntlProvider
  locale="en-US"
  messages={
    Object {
      "Terra.alert.advisory": "Advisory.",
      "Terra.alert.alert": "Alert.",
      "Terra.alert.dismiss": "Dismiss",
      "Terra.alert.error": "Error.",
      "Terra.alert.info": "Information.",
      "Terra.alert.success": "Success.",
      "Terra.alert.warning": "Warning.",
    }
  }
>
  <Alert
    action={null}
    customIcon={null}
    customStatusColor=""
    onDismiss={null}
    title=""
    type="success"
  >
    <ResponsiveElement
      defaultElement={
        <div
          className="alert-base success narrow"
          style={Object {}}
        >
          <div
            className="body body-std"
          >
            <span
              className="icon"
            >
              <IconSuccess
                viewBox="0 0 48 48"
                xmlns="http://www.w3.org/2000/svg"
              />
            </span>
            <div
              className="section"
            >
              <strong
                className="title"
              >
                Success.
              </strong>
              This is a success alert.
            </div>
          </div>
          
        </div>
      }
      responsiveTo="parent"
      tiny={
        <div
          className="alert-base success wide"
          style={Object {}}
        >
          <div
            className="body body-std"
          >
            <span
              className="icon"
            >
              <IconSuccess
                viewBox="0 0 48 48"
                xmlns="http://www.w3.org/2000/svg"
              />
            </span>
            <div
              className="section"
            >
              <strong
                className="title"
              >
                Success.
              </strong>
              This is a success alert.
            </div>
          </div>
          
        </div>
      }
    >
      <div />
    </ResponsiveElement>
  </Alert>
</IntlProvider>
`;

exports[`Alert of type warning with text content should render an Alert component of type warning 1`] = `
<IntlProvider
  locale="en-US"
  messages={
    Object {
      "Terra.alert.advisory": "Advisory.",
      "Terra.alert.alert": "Alert.",
      "Terra.alert.dismiss": "Dismiss",
      "Terra.alert.error": "Error.",
      "Terra.alert.info": "Information.",
      "Terra.alert.success": "Success.",
      "Terra.alert.warning": "Warning.",
    }
  }
>
  <Alert
    action={null}
    customIcon={null}
    customStatusColor=""
    onDismiss={null}
    title=""
    type="warning"
  >
    <ResponsiveElement
      defaultElement={
        <div
          className="alert-base warning narrow"
          style={Object {}}
        >
          <div
            className="body body-std"
          >
            <span
              className="icon"
            >
              <IconWarning
                viewBox="0 0 48 48"
                xmlns="http://www.w3.org/2000/svg"
              />
            </span>
            <div
              className="section"
            >
              <strong
                className="title"
              >
                Warning.
              </strong>
              This is an warning.
            </div>
          </div>
          
        </div>
      }
      responsiveTo="parent"
      tiny={
        <div
          className="alert-base warning wide"
          style={Object {}}
        >
          <div
            className="body body-std"
          >
            <span
              className="icon"
            >
              <IconWarning
                viewBox="0 0 48 48"
                xmlns="http://www.w3.org/2000/svg"
              />
            </span>
            <div
              className="section"
            >
              <strong
                className="title"
              >
                Warning.
              </strong>
              This is an warning.
            </div>
          </div>
          
        </div>
      }
    >
      <div />
    </ResponsiveElement>
  </Alert>
</IntlProvider>
`;

exports[`Alert with no props should render a default component 1`] = `
<IntlProvider
  locale="en-US"
  messages={
    Object {
      "Terra.alert.advisory": "Advisory.",
      "Terra.alert.alert": "Alert.",
      "Terra.alert.dismiss": "Dismiss",
      "Terra.alert.error": "Error.",
      "Terra.alert.info": "Information.",
      "Terra.alert.success": "Success.",
      "Terra.alert.warning": "Warning.",
    }
  }
>
  <Alert
    action={null}
    customIcon={null}
    customStatusColor=""
    onDismiss={null}
    title=""
    type="alert"
  >
    <ResponsiveElement
      defaultElement={
        <div
          className="alert-base alert narrow"
          style={Object {}}
        >
          <div
            className="body body-std"
          >
            <span
              className="icon"
            >
              <IconAlert
                viewBox="0 0 48 48"
                xmlns="http://www.w3.org/2000/svg"
              />
            </span>
            <div
              className="section"
            >
              <strong
                className="title"
              >
                Alert.
              </strong>
              
            </div>
          </div>
          
        </div>
      }
      responsiveTo="parent"
      tiny={
        <div
          className="alert-base alert wide"
          style={Object {}}
        >
          <div
            className="body body-std"
          >
            <span
              className="icon"
            >
              <IconAlert
                viewBox="0 0 48 48"
                xmlns="http://www.w3.org/2000/svg"
              />
            </span>
            <div
              className="section"
            >
              <strong
                className="title"
              >
                Alert.
              </strong>
              
            </div>
          </div>
          
        </div>
      }
    >
      <div />
    </ResponsiveElement>
  </Alert>
</IntlProvider>
`;

exports[`Dismissable Alert of type custom with action button, custom title and text content should render an Alert component of type custom with an action button 1`] = `
<IntlProvider
  locale="en-US"
  messages={
    Object {
      "Terra.alert.advisory": "Advisory.",
      "Terra.alert.alert": "Alert.",
      "Terra.alert.dismiss": "Dismiss",
      "Terra.alert.error": "Error.",
      "Terra.alert.info": "Information.",
      "Terra.alert.success": "Success.",
      "Terra.alert.warning": "Warning.",
    }
  }
>
  <Alert
    action={
      <Button
        isBlock={false}
        isCompact={false}
        isDisabled={false}
        isIconOnly={false}
        isReversed={false}
        onClick={[Function]}
        text="Action"
        type="button"
        variant="emphasis"
      />
    }
    customIcon={
      <IconHelp
        viewBox="0 0 48 48"
        xmlns="http://www.w3.org/2000/svg"
      />
    }
    customStatusColor="orange"
    onDismiss={[Function]}
    title="Help!"
    type="custom"
  >
    <ResponsiveElement
      defaultElement={
        <div
          className="alert-base custom narrow"
          style={
            Object {
              "color": "orange",
            }
          }
        >
          <div
            className="body body-narrow"
          >
            <span
              className="icon"
            >
              <IconHelp
                viewBox="0 0 48 48"
                xmlns="http://www.w3.org/2000/svg"
              />
            </span>
            <div
              className="section section-custom"
            >
              <strong
                className="title"
              >
                Help!
              </strong>
              This is a custom alert.
            </div>
          </div>
          <div
            className="actions actions-custom"
          >
            <Button
              isBlock={false}
              isCompact={false}
              isDisabled={false}
              isIconOnly={false}
              isReversed={false}
              onClick={[Function]}
              text="Action"
              type="button"
              variant="emphasis"
            />
            <Button
              isBlock={false}
              isCompact={false}
              isDisabled={false}
              isIconOnly={false}
              isReversed={false}
              onClick={[Function]}
              text="Dismiss"
              type="button"
              variant="neutral"
            />
          </div>
        </div>
      }
      responsiveTo="parent"
      tiny={
        <div
          className="alert-base custom wide"
          style={
            Object {
              "color": "orange",
            }
          }
        >
          <div
            className="body body-std"
          >
            <span
              className="icon"
            >
              <IconHelp
                viewBox="0 0 48 48"
                xmlns="http://www.w3.org/2000/svg"
              />
            </span>
            <div
              className="section section-custom"
            >
              <strong
                className="title"
              >
                Help!
              </strong>
              This is a custom alert.
            </div>
          </div>
          <div
            className="actions actions-custom"
          >
            <Button
              isBlock={false}
              isCompact={false}
              isDisabled={false}
              isIconOnly={false}
              isReversed={false}
              onClick={[Function]}
              text="Action"
              type="button"
              variant="emphasis"
            />
            <Button
              isBlock={false}
              isCompact={false}
              isDisabled={false}
              isIconOnly={false}
              isReversed={false}
              onClick={[Function]}
              text="Dismiss"
              type="button"
              variant="neutral"
            />
          </div>
        </div>
      }
    >
      <div />
    </ResponsiveElement>
  </Alert>
</IntlProvider>
`;

exports[`Dismissible Alert that includes actions section should render an alert component with a dismiss button 1`] = `
<IntlProvider
  locale="en-US"
  messages={
    Object {
      "Terra.alert.advisory": "Advisory.",
      "Terra.alert.alert": "Alert.",
      "Terra.alert.dismiss": "Dismiss",
      "Terra.alert.error": "Error.",
      "Terra.alert.info": "Information.",
      "Terra.alert.success": "Success.",
      "Terra.alert.warning": "Warning.",
    }
  }
>
  <Alert
    action={null}
    customIcon={null}
    customStatusColor=""
    onDismiss={[Function]}
    title=""
    type="alert"
  >
    <ResponsiveElement
      defaultElement={
        <div
          className="alert-base alert narrow"
          style={Object {}}
        >
          <div
            className="body body-narrow"
          >
            <span
              className="icon"
            >
              <IconAlert
                viewBox="0 0 48 48"
                xmlns="http://www.w3.org/2000/svg"
              />
            </span>
            <div
              className="section"
            >
              <strong
                className="title"
              >
                Alert.
              </strong>
              This is a test
            </div>
          </div>
          <div
            className="actions"
          >
            <Button
              isBlock={false}
              isCompact={false}
              isDisabled={false}
              isIconOnly={false}
              isReversed={false}
              onClick={[Function]}
              text="Dismiss"
              type="button"
              variant="neutral"
            />
          </div>
        </div>
      }
      responsiveTo="parent"
      tiny={
        <div
          className="alert-base alert wide"
          style={Object {}}
        >
          <div
            className="body body-std"
          >
            <span
              className="icon"
            >
              <IconAlert
                viewBox="0 0 48 48"
                xmlns="http://www.w3.org/2000/svg"
              />
            </span>
            <div
              className="section"
            >
              <strong
                className="title"
              >
                Alert.
              </strong>
              This is a test
            </div>
          </div>
          <div
            className="actions"
          >
            <Button
              isBlock={false}
              isCompact={false}
              isDisabled={false}
              isIconOnly={false}
              isReversed={false}
              onClick={[Function]}
              text="Dismiss"
              type="button"
              variant="neutral"
            />
          </div>
        </div>
      }
    >
      <div />
    </ResponsiveElement>
  </Alert>
</IntlProvider>
`;
